src/lib/encoder.ts

```diff
@@ -82,20 +80,22 @@ export const encodeDataSourceWithHash = (
   );
   return [
     '0x0000',
-    stripHexPrefix(
-      verificationMethod
-        ? padLeft(keccak256(verificationMethod.name).slice(0, 10), 8)
-        : '00000000',
-    ),
-    stripHexPrefix(
-      verification?.data
-        ? padLeft(verification.data.slice(2).length / 2, 4)
-        : '0000',
-    ),
-    stripHexPrefix(
-      verification?.data ? stripHexPrefix(verification?.data) : '',
-    ),
-    stripHexPrefix(utf8ToHex(dataSource)),
+    (verificationMethod
+      ? pad(keccak256(toBytes(verificationMethod.name)).slice(0, 10) as Hex, {
```

Comment:
Suggest using the `slice` function from viem here

```suggestion
      ? pad(slice(keccak256(toBytes(verificationMethod.name)), 0, 4), {
```

https://viem.sh/docs/utilities/slice

<img width="1098" alt="image" src="https://github.com/user-attachments/assets/6536a01e-83e7-429d-9330-1212eda94fed" />

Reply:
Also this line is very big and hard to read:

```ts
const verificationNameAsUtf8Hex = toBytes(verificationMethod.name);

const verificationMethodId = slice(keccak256(verificationNameAsUtf8Hex), 0, 4));
```

Reply:
I am also not sure why here we are padding on the left. Why is padding needed? Since we are taking just the first 4 bytes of the hash digest

Reply:
The padding was for 0x0 but you're right most of the time it's 0x00000000.
Fixed.